discovery_engine/lib/src/infrastructure/assets/native/data_provider.dart

```diff
@@ -65,36 +67,55 @@ class NativeDataProvider extends DataProvider {
   Future<String> _getData(Asset asset) async {
     final filePath =
         DataProvider.joinPaths([storageDirectoryPath, asset.urlSuffix]);
-    final assetFile = File(filePath);
-    final diskDirPath = assetFile.parent.path;
-    await Directory(diskDirPath).create(recursive: true);
-
-    // Only write the data on disk if the file does not exist or the checksum does not match.
-    // The last check is useful in case the app is closed before we can finish to write,
-    // and it can be also useful during development to test with different models.
-    var doesExist = assetFile.existsSync();
-
-    if (doesExist &&
-        !await _verifyChecksum(assetFile, asset.checksum.checksumAsHex)) {
-      await assetFile.delete();
-      doesExist = false;
-    }
+    final assetFileRef = File(filePath);
+    await Directory(assetFileRef.parent.path).create(recursive: true);
 
-    if (!doesExist) {
-      final bytes = await assetFetcher.fetchAsset(
-        asset,
-        onFetched: assetReporter.assetFetched,
-      );
-      await assetFile.writeAsBytes(bytes, flush: true);
-    } else if (asset.fragments.isEmpty) {
-      assetReporter.assetFetched(asset.urlSuffix);
-    } else {
+    // if the file exists it means it was verified and has a proper checksum
+    if (assetFileRef.existsSync()) {
+      if (asset.fragments.isEmpty) {
```

Comment:
If the file is in the proper place we should not care about the fragments. Is because how the the `assetReporter` works that we need to to send an event for each fragment anyway?

Reply:
the AssetReporter sums all files or if the file is splitted all its fragments, and uses that info to calculate the progress, so if the asset has fragments, we need to report on each fragment.